app/controllers/admin/api_controller.rb

```diff
@@ -0,0 +1,54 @@
+class Admin::ApiController < ApplicationController
+  before_filter :admin_only
+  before_filter :check_for_cancel, only: [:create, :update]
+
+  def index
```

Comment:
You know, funnily enough, I was looking at this last night and thinking the exact same thing about implementing a search. As you say, it shouldn't be difficult to implement, so I'll do that before this goes live.

I also agree that auditing (and maybe banning/throttling abilities) will be essential once we let anyone who isn't Open Doors use this feature. However, as I was also going to say before James commented, there may be better places to do that than in the app code/database, so this will require some further design/architectural discussion. I'll start an email thread about it so we can see what the options are.